src/czar/HttpCzarIngestModuleBase.cc

```diff
@@ -0,0 +1,374 @@
+    string const userDatabaseNamesPrefix = "user_";
+    if ((databaseName.size() <= userDatabaseNamesPrefix.size()) ||
+        !boost::iequals(databaseName.substr(0, userDatabaseNamesPrefix.size()), userDatabaseNamesPrefix)) {
+        auto err = "database name doesn't start with the prefix: " + userDatabaseNamesPrefix;
+        throw http::Error(func, err);
+    }
+}
+
+void HttpCzarIngestModuleBase::verifyUserTableName(string const& func, string const& tableName) const {
+    string const qservTableNamesPrefix = "qserv";
+    if (tableName.empty()) throw http::Error(func, "table name is empty");
+    if (boost::iequals(tableName.substr(0, qservTableNamesPrefix.size()), qservTableNamesPrefix)) {
+        auto err = "table name starts with the reserved prefix: " + qservTableNamesPrefix;
+        throw http::Error(func, err);
+    }
+}
+
+void HttpCzarIngestModuleBase::deleteDatabase(string const& databaseName) {
+    json data = json::object();
+    _requestController(http::Method::DELETE, "/ingest/database/" + databaseName, data);
+}
+
+void HttpCzarIngestModuleBase::deleteTable(string const& databaseName, string const& tableName) {
+    json data = json::object();
+    _requestController(http::Method::DELETE, "/ingest/table/" + databaseName + "/" + tableName, data);
+}
+
+vector<string> HttpCzarIngestModuleBase::getWorkerIds() {
+    vector<string> workerIds;
+    auto const workersJson = _requestController(http::Method::GET, "/replication/config");
+    for (auto const& worker : workersJson.at("config").at("workers")) {
+        bool const isEnabled = worker.at("is-enabled").get<int>() != 0;
+        bool const isReadOnly = worker.at("is-read-only").get<int>() != 0;
+        if (isEnabled && !isReadOnly) {
+            workerIds.push_back(worker.at("name").get<string>());
+        }
+    }
+    if (workerIds.empty()) {
+        throw http::Error(__func__, "no workers found in this Qserv instance");
+    }
+    return workerIds;
+}
+
+void HttpCzarIngestModuleBase::_unpublishOrCreateDatabase(const string& databaseName) {
+    json const config = _requestController(http::Method::GET, "/replication/config").at("config");
+    for (const auto& database : config.at("databases")) {
+        if (boost::iequals(database.at("database").get<string>(), databaseName)) {
+            if (database.at("is_published").get<int>() != 0) _unpublishDatabase(databaseName);
+            if (::countDirectors(database) == 0) _createDirectorTable(databaseName);
+            return;
+        }
+    }
+    _createDatabase(databaseName);
+    _createDirectorTable(databaseName);
+}
+
+void HttpCzarIngestModuleBase::_createDatabase(string const& databaseName) {
+    json data = json::object({{"database", databaseName},
+                              {"num_stripes", ::defaultNumStripes},
+                              {"num_sub_stripes", ::defaultNumSubStripes},
+                              {"overlap", ::defaultOverlap}});
```

Comment:
The comment at definition of these values leads me to believe they have no real effect here. Maybe they should have have values that indicate that? A replicated table effectively has 0 stripes, 0 substripes, and 0 overlap. If it's a partioned tables, these should be real values.